### PR TITLE
Report unhandled rejected promises

### PR DESCRIPTION
**Pull request by panuhorsmalahti**
Report unhandled rejected promises as per rsvp instructions.

### CODE REVIEW

**Comment by guybedford** [comment]
Thanks! I think we only need to do this once? Perhaps only in `cli.js` and `api.js`, the two entry points.

**Comment by panuhorsmalahti** [comment]
Yeah, technically it should be only required with the first require, but perhaps it's cleaner to do it always to be certain?

**Comment by guybedford** [comment]
Because it's an event we only need to do it once, otherwise we will have multiple errors.

**Comment by panuhorsmalahti** [comment]
Ah, that's a good point. Moved those to api.js and cli.js. I also cleaned up extra whitespace at the end of lines (it's my editor setting).

File: lib/core.js [hunk 1]
```diff
@@ -14,7 +14,8 @@
  *   limitations under the License.
  */
 
-var Promise = require('rsvp').Promise;
+var RSVP = require('rsvp');
+var Promise = RSVP.Promise;
```
Comment (guybedford) [inline lib/core.js]: This isn't necessary.

File: lib/install.js [hunk 2]
```diff
@@ -1,5 +1,6 @@
 var config = require('./config');
-var Promise = require('rsvp').Promise;
+var RSVP = require('rsvp');
+var Promise = RSVP.Promise;
```
Comment (guybedford) [inline lib/install.js]: Again, not necessary.

**Comment by guybedford** [comment]
Sure thanks, it would be nicer without the whitespace diffs though.

Also, the Promise requires don't need to be changed anymore.

**Comment by panuhorsmalahti** [comment]
Changed the requires.

**Comment by guybedford** [comment]
Great thanks, can you squash the commits?

**Comment by panuhorsmalahti** [comment]
Squashed commits.

**Comment by guybedford** [comment]
I was thinking it may make more sense for this to be a soft failure instead
of crashing everything:

```javascript
ui.log('warn', 'Unhandled rejection \n' + err.stack || err);
```

**Comment by panuhorsmalahti** [comment]
It's better to close the process.

**Comment by guybedford** [comment]
I'm not sure? I don't want to crash an application when it doesn't need to. There is one valid case that unhandled rejections might happen, which is if a fork resolution results in an existing download no longer being relevant. In such a scenario, the download promise is no longer listened to, and may no longer lead to a rejection handler, but that wouldn't be critical.

**Comment by panuhorsmalahti** [comment]
If you don't want the application to crash you should handle the rejection.

**Comment by guybedford** [comment]
@panuhorsmalahti this would involve adding listeners during the point of fork resolution to ensure that unattached download promises have silent rejections.

**Comment by guybedford** [comment]
@panuhorsmalahti thanks so much for driving this, I've separately merged a first implementation in https://github.com/jspm/jspm-cli/commit/fe6af8799d42c827d1f79f6888b86c89cb3e6439.

**Pull request closed without merging**